module-4/backend/week_one.md

```diff
@@ -1,20 +1,94 @@
-## Week One - Module 4 Recap
-
-Fork this respository. Answer the questions to the best of your ability. Try to answer them with limited amount of external research. These questions cover the majority of what we've learned this week. 
-
-Note: When you're done, submit a PR. 
-
-1. What's the most useful thing you learned from completing the intermission week work?
-2. What are some tools to help debug JavaScript code?
-3. What are some tools you need in order to unit test your JavaScript?
-4. What is the syntax for invoking a function?
-5. What's the difference between `==` and `===` in JavaScript?
-6. What's the difference between asynchronous and synchronous JavaScript? 
-7. How do we setup a route when creating an API with Node and Express?
-8. What do we use Knex for?
-9. What's `npm` and what do we use it for?
-
-#### Review  
-10. What's the MVC design pattern? Describe each part of MVC?
-11. What is AJAX? What are some benefits of using AJAX?
-12. What's a background worker? When would we want to use a background worker?
+# Week One - Module 4 Recap
+
+Fork this repository. Answer the questions to the best of your ability. Try to answer them with limited amount of external research. These questions cover the majority of what we've learned this week.
+
+Note: When you're done, submit a PR.
+
+### What's the most useful thing you learned from completing the intermission week work?
+
+The intermission work itself was underwhelming. I worked through six JS exercisms and the first 10 of Wes Bos' "Javascript 30" course, found it very helpful. Perhaps include it in future week's pre-work?
+
+Our first two days of JS in class could have been pre-work too.
+
+This would make for quite a bit more pre-work, but I would have preferred that.
+
+### What are some tools to help debug JavaScript code?
+
+PryJS, 'debugger'.
+
+### What are some tools you need in order to unit test your JavaScript?
+
+mocha and chai
+
+### What is the syntax for invoking a function?
+
+```javascript
+myCoolFunction()
+```
+
+### What's the difference between == and === in JavaScript?
+
+`==` is an equality operator
+`===` is "deep equality", and useful when `==` fails. (I don't recall more than this. I always reach for `==`)
+
+### What's the difference between asynchronous and synchronous JavaScript?
+
+most JS is synch. That's why for tests you're always adding `function(done)` all over, to let the whole test run before the test finishes.
+
+AJAX is asynch, and lets you keep "open" functions while waiting for stuff to finish.
+
+Asynch JS keeps you out of callback hell, along with promises (`.then()`)
```

Comment:
Callback hell is actually related to asynchronous functions, but you're right about `.then()`.